central/notifier/processor/singleton.go

```diff
@@ -88,17 +88,19 @@ func initialize() {
 			if secured && !needsRekey {
 				continue
 			}
-			if needsRekey {
-				err = notifierUtils.RekeyNotifier(protoNotifier, oldKey, cryptoKey)
+
+			if !secured {
```

Comment:
Isn't this always true when we reache here - given the continue condition above?

Reply:
The If condition above checks `secured || !needsRekey`. 
```
if secured && !needsRekey {
    continue
}
```
So `secured = false` and `needsRekey = true` would make if condition false and code below it would be executed. In this case, we would want to simply encrypt that notifier rather than rekey. That is why I flipped the check to give preference to securing over rekeying.

Reply:
It might be better to separate out the rekeying path?